src/commands/init.js

```diff
@@ -14,206 +14,241 @@ const { track } = require('../utils/telemetry')
 
-This will allow for Netlify Continuous deployment to build branch & PR previews.
+  persistState({ state, siteInfo })
 
-For more details on Netlify CI checkout the docs: http://bit.ly/2N0Jhy5
-`)
-      if (repo.error === "Couldn't find origin url") {
-        this.log(`Unable to find a remote origin URL. Please add a git remote.
+  context.exit()
+}
 
-git remote add origin https://github.com/YourUserName/RepoName.git
-`)
-      }
-
-      const NEW_SITE_NO_GIT = 'Yes, create and deploy site manually'
-      const NO_ABORT = 'No, I will connect this directory with GitHub first'
-
-      const { noGitRemoteChoice } = await inquirer.prompt([
-        {
-          type: 'list',
-          name: 'noGitRemoteChoice',
-          message: 'Do you want to create a Netlify site without a git repository?',
-          choices: [NEW_SITE_NO_GIT, NO_ABORT],
-        },
-      ])
-
-      // create site or search for one
-      if (noGitRemoteChoice === NEW_SITE_NO_GIT) {
-        // run site:create command
-        siteInfo = await SitesCreateCommand.run([])
-
-        this.log(`"${siteInfo.name}" site was created`)
-        this.log()
-        this.log(`To deploy to this site. Run your site build and then ${chalk.cyanBright.bold('netlify deploy')}`)
-
-        // Save to .netlify/state.json file
-        state.set('siteId', siteInfo.id)
-
-        // no github remote
-        this.exit()
-      } else if (noGitRemoteChoice === NO_ABORT) {
-        this.log()
-        this.log(`${chalk.bold('To initialize a new git repo follow the steps below.')}
+const logGitSetupInstructionsAndExit = ({ context }) => {
+  context.log()
+  context.log(`${chalk.bold('To initialize a new git repo follow the steps below.')}
 
 1. Initialize a new repo:
 
-   ${chalk.cyanBright.bold('git init')}
+${chalk.cyanBright.bold('git init')}
 
 2. Commit your files
 
-   ${chalk.cyanBright.bold('git add .')}
+${chalk.cyanBright.bold('git add .')}
 
 3. Commit your files
 
-   ${chalk.cyanBright.bold("git commit -m 'initial commit'")}
+${chalk.cyanBright.bold("git commit -m 'initial commit'")}
 
 4. Create a new repo in GitHub ${chalk.cyanBright.bold('https://github.com/new')}
 
 5. Link the remote repo with this local directory
 
-   ${chalk.cyanBright.bold('git remote add origin git@github.com:YourGithubName/your-repo-slug.git')}
+${chalk.cyanBright.bold('git remote add origin git@github.com:YourGithubName/your-repo-slug.git')}
 
 6. Push up your files
 
-   ${chalk.cyanBright.bold('git push -u origin master')}
+${chalk.cyanBright.bold('git push -u origin master')}
 
 7. Initialize your Netlify Site
 
-   ${chalk.cyanBright.bold('netlify init')}
+${chalk.cyanBright.bold('netlify init')}
 `)
-        this.exit()
-      }
+  context.exit()
+}
 
-      // Throw github remote error
-      this.error(repo.error)
-    }
+const handleNoGitRemoteAndExit = async ({ context, error, state }) => {
+  context.log()
```

Comment:
Unrelated to this PR, but I was wondering: do you know why we are passing the `log` function around instead of importing it in the top-level scope?

Reply:
Not sure what you mean by this. We use the `log` method from the `Command` class - we even override it:
https://github.com/netlify/cli/blob/6ea7ab428ce88c9ce4ea682eff8799777672efae/src/utils/command.js#L137

Reply:
Yes, exactly.
We should extract this `log` class method into a normal function instead. Like this, we don't need to pass the `context` object around just due to `log()`. What do you think?

Reply:
Opened https://github.com/netlify/cli/issues/1615 to continue the discussion there